.gitignore

```diff
@@ -50,3 +50,4 @@ third_party/zlib/minigzipsh
 third_party/zlib/zconf.h
 third_party/zlib/zlib.pc
 third_party/zlib/Makefile
+third_party*
```

Comment:
@ravic-rs any opinion about completely ignoring the third_party dir as a project default? I'm not sure if this would complicate developers adding new elements to the third_party dir in the future or not. Seems like it could lead to a developer not realizing they have third_party files they need to stage (assuming they added some). 

On second look, there isn't a slash between third_party and * so i guess this is just ignoring any folder named third_party*, but that still seems like an unneeded change here.

Reply:
Agreed on those concerns.
@zaryabkhan-rs please remove this line. During discussion, it was meant for local .gitignore to aid it viewing/staged changes.